### PR TITLE
chore(jenkins): run Jenkins builds on Node 4 (via nvm)

### PR DESCRIPTION
**Pull request by petebacondarwin**

### CODE REVIEW

File: scripts/jenkins/set-node-version.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+# Install nvm for this shell
+source ~/.nvm/nvm.sh
```
Comment (Narretz) [inline scripts/jenkins/set-node-version.sh]: Does jenkins provide this?

Comment (petebacondarwin) [inline scripts/jenkins/set-node-version.sh]: It is manually installed on the server.

File: scripts/jenkins/master.sh [hunk 2]
```diff
@@ -17,14 +17,7 @@ function init {
 
 function build {
   cd ../..
-
-  if [[ $NO_TEST == "true" ]]; then
```
Comment (gkalpak) [inline scripts/jenkins/master.sh]: If the `--no-test` option is not used any more, should we remove it from `ARG_DEFS` [above][1] ?

[1]: https://github.com/angular/angular.js/pull/13568/files#diff-8ba5664b63cf24ac7a492e66ec3364fcR7

File: scripts/jenkins/set-node-version.sh [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+
+# Use node.js at 4.2.x
+nvm install 4.2
```
Comment (gkalpak) [inline scripts/jenkins/set-node-version.sh]: Is `nvm install ...` sufficient or do we also need `nvm use ...` ?

File: scripts/jenkins/master.sh (hunk 2 above)
Comment (petebacondarwin) [inline scripts/jenkins/master.sh]: Oh yes. Good spot

File: scripts/jenkins/set-node-version.sh (hunk 3 above)
Comment (petebacondarwin) [inline scripts/jenkins/set-node-version.sh]: `nvm install` is sufficient.

File: scripts/jenkins/master.sh (hunk 2 above)
Comment (petebacondarwin) [inline scripts/jenkins/master.sh]: Added at 4bcb307abc29d35fa1fa3a6fd3ab1ef39dd52576 (and similar commits in each of the other branches).

File: scripts/jenkins/set-node-version.sh (hunk 1 above)
Comment (mzgol) [inline scripts/jenkins/set-node-version.sh]: Could we print the nvm/Node/npm versions used? Could be useful for debugging.

File: scripts/jenkins/set-node-version.sh (hunk 3 above)
Comment (mzgol) [inline scripts/jenkins/set-node-version.sh]: `4` would be enough, the latest `4.x.y` release is an LTS, it so happens it's `4.2.y` in fact.

File: scripts/jenkins/set-node-version.sh (hunk 1 above)
Comment (petebacondarwin) [inline scripts/jenkins/set-node-version.sh]: I think that it does do that automatically...

File: scripts/jenkins/set-node-version.sh (hunk 3 above)
Comment (petebacondarwin) [inline scripts/jenkins/set-node-version.sh]: From the node.js docs:

> **Minors:** Minors are the addition and refinements of APIs or subsystems. They do not generally change APIs or introduce backwards compatible breaking changes, except where unavoidable. These are mostly completely additive releases.

I think it is safer to stick with a fixed minor release for automatic updates

Comment (mzgol) [inline scripts/jenkins/set-node-version.sh]: OK. It doesn't matter here anyway as there will be no Node.js 4.3.

Comment (petebacondarwin) [inline scripts/jenkins/set-node-version.sh]: I see: https://github.com/nodejs/LTS/

**Pull request closed without merging**